src/main/java/com/artipie/api/ManageUsers.java

```diff
@@ -109,7 +119,40 @@ public void remove(final String uid) {
             );
             return;
         }
-        throw new IllegalStateException(String.format("User %s does not exist", uid));
+        throw new IllegalStateException(String.format("User %s does not exist", uname));
+    }
+
+    @Override
+    public void alterPassword(final String uname, final JsonObject info) {
+        if (this.users().map(yaml -> yaml.yamlMapping(uname) != null).orElse(false)) {
+            YamlMappingBuilder all = Yaml.createYamlMappingBuilder();
+            final YamlMapping users = this.users().get();
+            for (final YamlNode node : users.keys()) {
+                final String val = node.asScalar().value();
+                if (!uname.equals(val)) {
+                    all = all.add(val, users.yamlMapping(val));
+                }
+            }
+            final YamlMapping user = this.users().get().yamlMapping(uname);
+            YamlMappingBuilder changing = Yaml.createYamlMappingBuilder();
+            for (final YamlNode node : user.keys()) {
+                final String val = node.asScalar().value();
+                if (!ManageUsers.TYPE.equals(val) && !ManageUsers.PASS.equals(val)) {
+                    changing = changing.add(val, user.value(val));
+                }
+            }
+            changing = changing.add(ManageUsers.PASS, info.getString("new_pass"));
+            changing = changing.add(ManageUsers.TYPE, info.getString("new_type"));
+            all = all.add(uname, changing.build());
+            this.blsto.save(
+                this.key,
+                Yaml.createYamlMappingBuilder()
+                    .add(YamlSettings.NODE_CREDENTIALS, all.build())
+                    .build().toString().getBytes(StandardCharsets.UTF_8)
+            );
+            return;
+        }
+        throw new IllegalStateException(String.format("User %s does not found", uname));
```

Comment:
Cannot find usage of 'old_pass'

Reply:
check `UsersRest.java:128`, it is not a responsibility of `ManageUsers` class to verify current password, `UsersRest` does it before calling `ManageUsers.alterPassword()` to apply the changes